pkg/daemon/ceph/agent/flexvolume/manager/ceph/manager.go

```diff
@@ -155,14 +174,30 @@ func (vm *VolumeManager) Detach(image, pool, clusterNamespace string, force bool
 		return nil
 	}
 
+	if id == "" && key == "" {
+		return fmt.Errorf("no id nor keyring given, can't unmount without credentials")
```

Comment:
ummapping requires the keyring?

Reply:
I need to verify, though the unmap command has the `keyring` flag set on it so I just added it to make sure the credentials are set as in the map command.